### ISSUE TITLE
Run as action on PR

### ISSUE THREAD

**Issue by zwass** [step-security/secure-workflows#583]
It would be awesome if this could be run as an action either nightly, or on PR, so that it could add github security scanning and/or code review suggestions to make the suggested improvements. I see there's #393 for crawling repos, but what if I just want to run it in my own repo?

**Comment by varunsh-coder** [comment]
Thanks @zwass for the suggestion! The [scorecard-action](https://github.com/ossf/scorecard-action) already scans for 2 (token permissions and pinned dependencies) of the 3 issues that this project fixes. In fact scorecard [suggests](https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions) this project to fix those two issues, and that is how developers typically become aware of it. I could discuss with scorecard maintainers to add suggested fixes as part of the scorecard-action run. That way developers do not need to add multiple workflows. What do you think?

**Comment by zwass** [comment]
Yes, the suggested fixes are what would really be awesome! Otherwise you have to open up the tool, manually copy the workflow, and then copy it back over and PR it.

**Comment by varunsh-coder** [comment]
@naveensrinivasan, @laurentsimon, @azeemshaikh38, what are your thoughts on integrating suggested fixes from this project into scorecard findings for token permissions and pinned dependencies? https://github.com/ossf/allstar could be another integration point, so adding @jeffmendoza. 

Other ideas also welcome on how to make it easier to suggest/ fix the workflows.

**Comment by azeemshaikh38** [comment]
I really like the idea of integrating this with AllStar. AllStar will soon start supporting Scorecard checks as policies, and if users have `fix` action enabled, AllStar can piggyback on this to provide an automated fix/suggestion.

@brianrussell2 @jeffmendoza wdyt?

**Comment by laurentsimon** [comment]
+1 this would be a wonderful feature. We can add a `Diff` to each details, and this can be picked up by AllStar, etc.
/cc @[sethvargo](https://github.com/sethvargo)

**Comment by jeffmendoza** [comment]
> AllStar will soon start supporting Scorecard checks as policies, and if users have fix action enabled, AllStar can piggyback on this to provide an automated fix/suggestion.

Yes, if the tool can provide an automated fix suggestion, allstar could open a PR with that suggestion if the user has the "fix" action configured.

**Comment by varunsh-coder** [comment]
Updated title. Tracking issue in scorecard - https://github.com/ossf/scorecard/issues/1850
